### ISSUE TITLE
Report line numbers and column numbers in parse error output

### ISSUE THREAD

**Issue by ievans** [semgrep/semgrep#746]
Very difficult to debug eg https://semgrep.live/Ov2 without line/col for parse errors

**Comment by rcoh** [comment]
And also https://semgrep.live/gQx

**Comment by rcoh** [comment]
#773 does most of the backend work for this. Some more QA and cleanups probably required afterwards.

**Comment by rcoh** [comment]
The line and column numbers are actually already in the JSON cc @DrewDennison

**Comment by rcoh** [comment]
I'm going to close this for `semgrep` -- all remaining work is in semgrep.live cc @DrewDennison @ievans

**Comment by DrewDennison** [comment]
@rcoh is this true if `--strict` is passed in? It wasn't last time I checked

**Comment by ievans** [comment]
#948 fixes this for code, but it is still a problem for the patterns:

```
(semgrep) ine@imbp4 ~/D/r/s/semgrep (ie/strict-die)> semgrep  --json --lang python --pattern '$X ==' foo.py | jq
non-zero return code while invoking semgrep-core:
	invalid pattern "$X ==": Parsing.Parse_error
An error occurred while invoking the semgrep engine; please help us fix this by creating an issue at https://semgrep.dev
```

**Comment by rcoh** [comment]
yeah, I'll have it fixed for patterns today. I thought the issue was just for code.

**Comment by dlukeomalley** [comment]
Has this issue been fixed? I've seen quite a bit of activity and believe this ticket is at worst "in progress". 🏡

**Comment by rcoh** [comment]
#998 gets pretty close

**Comment by dlukeomalley** [comment]
Sweet. I associated this ticket with #998 and we can close it out once that merges.

**Comment by rcoh** [comment]
Fixed in #1058
